File: paddler_client_python/paddler_client/generate_embedding_batch_params.py
from pydantic import BaseModel

from paddler_client.embedding_input_document import EmbeddingInputDocument
from paddler_client.embedding_normalization_method import (
    EmbeddingNormalizationMethod,
)


class GenerateEmbeddingBatchParams(BaseModel):
    input_batch: list[EmbeddingInputDocument]
    normalization_method: EmbeddingNormalizationMethod

File: paddler_client_javascript/src/schemas/GenerateEmbeddingBatchParams.ts
import { z } from "zod";

import { EmbeddingInputDocumentSchema } from "./EmbeddingInputDocument";
import { EmbeddingNormalizationMethodSchema } from "./EmbeddingNormalizationMethod";

export const GenerateEmbeddingBatchParamsSchema = z.object({
  input_documents: z.array(EmbeddingInputDocumentSchema),
  normalization_method: EmbeddingNormalizationMethodSchema,
});

export type GenerateEmbeddingBatchParams = z.infer<
  typeof GenerateEmbeddingBatchParamsSchema
>;
